Packs/CommonScripts/Scripts/SearchIncidentsV2/SearchIncidentsV2.py

```diff
@@ -163,24 +162,27 @@ def search_incidents(args: Dict):   # pragma: no cover
     demisto.debug(
         f'Amount of incidents after filtering = {len(all_found_incidents)} before pagination'
     )
-    # adding 1 here because the default page number start from 0
-    max_page = (res[0]["Contents"]["total"] // DEFAULT_PAGE_SIZE) + 1
-    demisto.debug(f'{max_page=}')
-
+    page_size = args.get('size') or DEFAULT_PAGE_SIZE
     page = STARTING_PAGE_NUMBER
-    while len(all_found_incidents) < limit and page < max_page:
+    more_pages = True
+    while more_pages and len(all_found_incidents) < limit:
         args['page'] = page
         current_page_found_incidents = execute_command('getIncidents', args).get('data') or []
-        demisto.debug(
-            f'before filtering {len(current_page_found_incidents)=} '
-            f' {args=} {page=}'
-        )
+
+        # When current_page_found_incidents is None it means the requested page was empty
+        if not current_page_found_incidents:
+            break
+
+        demisto.debug(f'before filtering {len(current_page_found_incidents)=} {args=} {page=}')
```

Comment:
would add a log for after filtering len too

Reply:
roger

Reply:
@JudahSchwartz but there is line 180